main_app/views.py

```diff
@@ -136,10 +168,46 @@ class LogoutPageView(LogoutView):
     pass
 
 
-class UserPageView(TemplateView):
+class UserPageView(LoginRequiredMixin, TemplateView):
     template_name = 'user_page.html'
 
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
         context['age'] = round((datetime.date(datetime.now()) - self.request.user.birthday).days / 365.25)
         return context
+
+
+class MustListView(LoginRequiredMixin, ListView):
+    template_name = 'must_page.html'
+    paginate_by = 10
+
+    def get_queryset(self):
+        # game_ids = [each.game_id for each in Must.objects.filter(user=self.request.user)]
+        queryset = Must.objects.filter(user=self.request.user).all()
+        # if game_ids:
+        #     queryset = Game.objects.filter(id__in=game_ids)
+        print(queryset[0].game)
+        if len(queryset):
+            return queryset
+        return []
+
+
+class AddRemoveMustView(View):
+    def post(self, request):
+        game_id = int(request.POST.get('game_id', 0))
+        add = request.POST.get('add')
+        force_remove = request.POST.get('force')
+        if game_id:
+            if add:
+                must = Must.objects.get_or_create(game=Game.objects.get(id=game_id), user=request.user)
+                must[0].is_deleted = False
+                must[0].save()
+            elif force_remove:
+                must = Must.objects.get(game=Game.objects.get(id=game_id), user=request.user)
+                must.delete()
+            else:
+                must = Must.objects.get(game=Game.objects.get(id=game_id), user=request.user)
+                must.is_deleted = True
+                must.save()
+        return HttpResponse('')
```

Comment:
```
    game_id = int(request.POST.get('game_id'))
    add = request.POST.get('add')
    game = Game.objects.get(id=game_id)
    if add:
        must, created = Must.objects.get_or_create(game=game, user=request.user)
        must.is_deleted = False
        must.save()
    else:
        Must.objects.filter(game=game, user=request.user).delete()
    return JsonResponse('status': 'ok')
```

We won't do force remove from UI. When I said that it would be good to have possibility for force remove I meant that we can override `delete()` method in model.
https://docs.djangoproject.com/en/2.2/topics/db/models/#overriding-predefined-model-methods
```
Class Must(models.Model):
.....
def delete(self, force=False, *args, **kwargs):
   if force:
        super().delete(*args, **kwargs)
   self.is_deleted = True
```